drivers: bluetooth: silabs_siwx91x: Add RF power config

Arunnagelly opened the pull request:
Add vendor-specific RF power mode configuration during HCI setup
for SiLabs SiWx91x devices. This configures the BLE TX power
index in the controller's RF subsystem.

The implementation sends a vendor command (OCF 0x0006) with
protocol mode (2) and power index (RSI_BLE_PWR_INX) during the
setup phase. This ensures proper RF power configuration before
normal Bluetooth operations begin.

Technical details:
- Uses bt_hci_cmd_alloc() with manual HCI header construction
- Command opcode: 0xFC06 (OGF=0x3F, OCF=0x0006)
- Parameters: protocol_mode=2, power_index from RSI_BLE_PWR_INX
- Executed during bt_hci_setup callback

jhedberg commented on drivers/bluetooth/hci/hci_silabs_siwx91x.c:
```diff
@@ -28,6 +56,7 @@ static int siwx91x_bt_open(const struct device *dev, bt_hci_recv_t recv)
 
 	if (!status) {
 		hci->recv = recv;
+		rsi_bt_driver_send_tx_pwr_vs_cmd(dev);
```
Actually, I see you're not using the host APIs for this. Which does make it possible to do this in open(). Might still be cleaner to use setup(), assuming that this is oral vendor command with usual HCI command semantics (eg resulting in a command complete event)

jhedberg commented on drivers/bluetooth/hci/hci_silabs_siwx91x.c:
```diff
@@ -5,21 +5,64 @@
+		net_buf_unref(rsp);
+	}
+
+	LOG_INF("RF Power Mode configured successfully");
```
This should be `LOG_DBG()`

jhedberg commented on drivers/bluetooth/hci/hci_silabs_siwx91x.c:
```diff
@@ -32,6 +75,17 @@ static int siwx91x_bt_open(const struct device *dev, bt_hci_recv_t recv)
 
+static int siwx91x_bt_setup(const struct device *dev, const struct bt_hci_setup_params *params)
+{
+    int ret = rsi_bt_driver_send_tx_pwr_vs_cmd(dev);
```
Please be consistent with error variable naming. You're using `err` in the other place, and that's also what I'd prefer here. Also, I think this will look a bit cleaner as:
```c
{
	int err;

	err = rsi_bt_driver_send_tx_pwr_vs_cmd(dev);
	if (err < 0) {
		...
```

The main point is that if you ever need to extend this with more commands then it's not very clean to do the first one immediately when you declare of the variable.

jhedberg commented on tests/net/ipv6/src/main.c:
```diff
@@ -848,7 +848,7 @@ ZTEST(net_ipv6, test_send_neighbor_discovery)
 
 	zassert_equal(atomic_get(&tx_data->avail_count), avail_buf_count,
-		      "Unexpected tx data pool available count (%d vs %d)",
+		      "Unexpected tx data pool available count (%ld vs %d)",
```
Seems like a completely unrelated change?

jhedberg commented on drivers/bluetooth/hci/hci_silabs_siwx91x.c:
```diff
@@ -5,21 +5,68 @@
+
+	/* Manually add HCI command header */
+	net_buf_add_le16(buf, BT_OP_VS_RF_POWER_MODE);
+	net_buf_add_u8(buf, sizeof(protocol_mode) + sizeof(le_tx_power_index));
```
This is not correct. `bt_hci_cmd_send_sync()` will insert these automatically into the buffer. Did you actually test this?

jhedberg replied:
It is still not correct. The controller will be receiving invalid (or at least unintended) values for the parameters, since it'll interpret the above bytes as parameters instead of the command header.

jhedberg reviewed (changes requested):
Starting to look pretty good to me. A few minor issues still.

jhedberg commented on drivers/bluetooth/hci/hci_silabs_siwx91x.c:
```diff
@@ -12,14 +12,57 @@
+	if (!buf) {
+		LOG_ERR("Failed to allocate HCI command buffer");
+		return -ENOMEM;
+	}
```
Why are you allocating the buffer twice? The second allocation also leaks the memory from the first one.

jerome-pouiller reviewed (dismissed):
I had several cosmetics comments in mind. However, this patch seems to fix the issue.

Arunnagelly committed:
drivers: bluetooth: hci: siwx91x: add RF power config

Add vendor-specific RF power mode configuration during HCI setup
for SiLabs SiWx91x devices. This configures the BLE TX power
index in the controller's RF subsystem.

The implementation sends a vendor command (OCF 0x0006) with
protocol mode (2) and power index (RSI_BLE_PWR_INX) during the
setup phase. This ensures proper RF power configuration before
normal Bluetooth operations begin.

Technical details:
- Uses bt_hci_cmd_alloc() with manual HCI header construction
- Command opcode: 0xFC06 (OGF=0x3F, OCF=0x0006)
- Parameters: protocol_mode=2, power_index from RSI_BLE_PWR_INX
- Executed during bt_hci_setup callback

Signed-off-by: Arun Kumar Nagelly <arnagell@silabs.com>
